py3status/module.py

```diff
@@ -416,6 +407,15 @@ def make_quotes(options):
                     raise ValueError(err)
                 self.py3status_module_options["position"] = position
 
+        markup = fn(self.module_full_name, "markup")
+        if not hasattr(markup, "none_setting"):
+            if markup not in MARKUP_LANGUAGES:
+                err = "Invalid `markup` attribute, should be "
+                err += make_quotes(MARKUP_LANGUAGES)
+                err += ". Got `{}`.".format(markup)
+                raise ValueError(err)
+            self.py3status_module_options["markup"] = markup
```

Comment:
can you explain why moving it down is necessary? sorry not very familiar with this part of codebase

Reply:
```diff
diff --git a/py3status/module.py b/py3status/module.py
index 2ab7595..56e0133 100644
--- a/py3status/module.py
+++ b/py3status/module.py
@@ -347,7 +347,7 @@ class Module:
                 err += make_quotes(MARKUP_LANGUAGES)
                 err += ". Got `{}`.".format(markup)
                 raise ValueError(err)
-            self.i3bar_module_options["markup"] = markup
+            self.py3status_module_options["markup"] = markup
 
         separator = fn(self.module_full_name, "separator")
         if not hasattr(separator, "none_setting"):
```

No real change. I moved it down to be with other `py3status` options (`min_length` and `positions`) as it would be our code (i.e. `py3status`) to deal with `markup` for composites instead of just pushing everything out as-is `i3bar`.